### DISCUSSION TITLE
Customising xavier nx boot order

### DISCUSSION THREAD

**Discussion by DavidJRobertson** [OE4T/meta-tegra#890] [Q&A] [answered]
How do I change the cboot boot order on the Xavier NX? 

On the AGX Xavier (not using OE4T) I used a script like below to do it. I assume that means I need to do something similar with creating a cbo.dtb but I am not really sure where to put my cbo.dts to have it built and included in the tegraflash image.

```sh
dtc -I dts -O dtb -o cbo.dtb << EOF
	/dts-v1/;
	/ {
		compatible = "nvidia,cboot-options-v1";
		boot-configuration {
			boot-order = "sd", "emmc", "nvme", "usb";
		};
	};
EOF

dd if=cbo.dtb of=/dev/disk/by-partlabel/CPUBL-CFG 
```

**Comment by madisongh** [comment, marked as answer]
It's a two-step process.  The first step is to add a recipe with your `cbo.dts` file.  Next is to add a bbappend
for the `tegra-bootfiles` recipe to have it depend on your new recipe and set CBOOTOPTION_FILE to point
to the file.  I have an example in my test distro repo: [recipe to install the file](https://github.com/madisongh/tegra-test-distro/blob/master/layers/meta-testdistro/recipes-bsp/tegra-binaries/bootfiles_1.0.bb), [bbappend for tegra-bootfiles](https://github.com/madisongh/tegra-test-distro/blob/master/layers/meta-testdistro/recipes-bsp/tegra-binaries/tegra-bootfiles_32.6.1.bbappend)

**Reply by DavidJRobertson** [reply]
Thanks, didn't see that. Got it working now.
